src/machine.rs

```diff
@@ -546,35 +547,29 @@ impl OlmMachine {
     /// **Warning**: Only export this and share it with a trusted recipient,
     /// i.e. if an existing device is sharing this with a new device.
     #[wasm_bindgen(js_name = "exportSecretsBundle")]
-    pub fn export_secrets_bundle(&self) -> Promise {
-        let me = self.inner.clone();
-
-        future_to_promise(async move {
-            Ok(me.store().export_secrets_bundle().await.map(store::SecretsBundle::from)?)
-        })
+    pub async fn export_secrets_bundle(&self) -> Result<store::SecretsBundle, JsError> {
+        Ok(self.inner.store().export_secrets_bundle().await?.into())
     }
 
-    /// Import and persists secrets from a [`SecretsBundle`].
+    /// Import and persists secrets from a {@link SecretsBundle}.
     ///
     /// This method will import all the private cross-signing keys and, if
     /// available, the private part of a backup key and its accompanying
     /// version into the store.
     ///
     /// **Warning**: Only import this from a trusted source, i.e. if an existing
     /// device is sharing this with a new device. The imported cross-signing
-    /// keys will create a [`OwnUserIdentity`] and mark it as verified.
+    /// keys will create a {@link OwnUserIdentity} and mark it as verified.
     ///
     /// The backup key will be persisted in the store and can be enabled using
-    /// the [`BackupMachine`].
+    /// the {@link BackupMachine}.
+    ///
+    /// The provided `SecretsBundle` is freed by this method; be careful not to
+    /// use it once this method has been called.
```

Comment:
invalidating an argument is a very non-JS thing to do, so let's call it out explicitly.